### ISSUE TITLE
Make Population.Epoch multithreaded

### ISSUE THREAD

**Issue by Psy-Kai** [peter-ch/MultiNEAT#38]
Is it possible to make the Population.Epoch call mutlithreaded? I played around with this library and found that this call is very slow. Especially the lines

    for(unsigned int i=0; i<m_Species.size(); i++)
    {
        m_Species[i].Reproduce(*this, m_Parameters, m_RNG);
    } 

Is it possible to call Species.Reproduce multithreadded (for every species in the list)?

**Comment by peter-ch** [comment]
Yeah, it can be multithreaded I guess. It just needs to be implemented.
